### PR TITLE
Add city and Country code on users

### PR DESCRIPTION
**Pull request by nronas**
closes: https://github.com/coronagoapp/coronago-backend/issues/10
## Why

We want to find out the country code and the city based on the lat, long pair.

## What

- Adds a new city column on the users table(no indexing yet, since we do not know the access patterns of this column)
- Adds [geocoder](https://github.com/alexreisner/geocoder) that provides reverse geocoding.

### Happy scenario
lat, lng: 51.509865, -0.118092 (expected behaviour, city: London, country_code: gb)

Screenshot 2020-03-28 at 18 05 58

### Failure scenario
Raising an exception while geocoding. Expected behaviour: Fail-safe on error

Screenshot 2020-03-28 at 18 06 54

/cc @prasadpilla @arunava59

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -28,6 +28,17 @@ class User < ApplicationRecord
...
   validates_length_of :name, minimum: 3, maximum: 30
 
+  # Reverse geocoding for city finder
+  reverse_geocoded_by :lat, :lng do |user, results|
+    if geo = results.first
+      user.city = geo.city
+      user.country_code = geo.country_code
+    end
+  rescue => e
+    Rails.logger.error("Failed to geocode city or country_code due to #{e.message}")
```
Comment (prasadpilla) [inline app/models/user.rb]: @nronas Can we log and report the exception(AppSignal)?
Analysing logs and implementing exception reports from there is an overkill for this app, so just directly reporting is probably better. 
We have couple of helpers to do that i think the mix is called ExceptionHelpers. 
ex: log_and_report(e)

Comment (prasadpilla) [reply app/models/user.rb]: Also, you can pass any context params to help debug the issue quicker.

Comment (nronas) [reply app/models/user.rb]: Had no clue that we use AppSignal. will amend

Comment (prasadpilla) [reply app/models/user.rb]: @nronas yeah, my fault. Didn't mention i anywhere, may be we should have a conventions guide somewhere

Comment (nronas) [reply app/models/user.rb]: no harm done

**Pull request merged**